src/openrct2/drawing/Text.cpp

```diff
@@ -0,0 +1,262 @@
+ *****************************************************************************/
+#pragma endregion
+
+#include "Text.h"
+
+extern "C"
+{
+    #include "../localisation/localisation.h"
+}
+
+static TextPaint _legacyPaint;
+
+void drawText(rct_drawpixelinfo *dpi, float x, float y, TextPaint *paint, utf8string text);
+void drawText(rct_drawpixelinfo *dpi, float x, float y, TextPaint *paint, rct_string_id format, void *args);
+
+StaticLayout::StaticLayout(utf8string source, TextPaint paint, sint32 width)
+{
+    _buffer = source;
+    _paint = paint;
+
+    sint32 fontSpriteBase;
+
+    gCurrentFontSpriteBase = paint.spriteBase;
+    _maxWidth = gfx_wrap_string(_buffer, width, &_lineCount, &fontSpriteBase);
+    _lineCount += 1;
+    _lineHeight = font_get_line_height(fontSpriteBase);
+}
+
+void StaticLayout::Draw(rct_drawpixelinfo *dpi, float x, float y)
+{
+    gCurrentFontFlags = 0;
+    gCurrentFontSpriteBase = _paint.spriteBase;
+
+    TextPaint tempPaint = _paint;
+
+    gCurrentFontFlags = 0;
+    sint32 lineY = y;
+    sint32 lineX = x;
+    switch (_paint.alignment) {
+        case TextAlignment::LEFT:
+            lineX = x;
+            break;
+        case TextAlignment::CENTRE:
+            lineX = x + _maxWidth / 2;
+            break;
+        case TextAlignment::RIGHT:
+            lineX = x + _maxWidth;
+            break;
+    }
+    utf8 *buffer = _buffer;
+    for (sint32 line = 0; line < _lineCount; ++line) {
+        drawText(dpi, lineX, lineY, &tempPaint, buffer);
+        tempPaint.colour = TEXT_COLOUR_254;
+        buffer = get_string_end(buffer) + 1;
+        lineY += _lineHeight;
+    }
+}
+
+sint32 StaticLayout::GetHeight()
+{
+    return _lineHeight * _lineCount;
+}
+
+sint32 StaticLayout::GetWidth()
+{
+    return _maxWidth;
+}
+
+sint32 StaticLayout::GetLineCount()
+{
+    return _lineCount;
+}
+
+
+void drawText(rct_drawpixelinfo *dpi, float x, float y, TextPaint *paint, utf8string text)
+{
+
+    sint32 width = gfx_get_string_width(text);
+
+    switch (paint->alignment) {
+        case TextAlignment::LEFT:
+            break;
+        case TextAlignment::CENTRE:
+            x -= width / 2;
+            break;
+        case TextAlignment::RIGHT:
+            x -= width;
+            break;
+    }
+
+    ttf_draw_string(dpi, text, paint->colour, x, y);
+
+    if (paint->underlineText) {
+        gfx_fill_rect(dpi, x, y + 11, x + width, y + 11, text_palette[1]);
+        if (text_palette[2] != 0)
+            gfx_fill_rect(dpi, x + 1, y + 12, x + width + 1, y + 12, text_palette[2]);
+    }
+}
+
+void drawText(rct_drawpixelinfo *dpi, float x, float y, TextPaint *paint, rct_string_id format, void *args)
+{
+
+    utf8 buffer[256];
+
+    format_string(buffer, sizeof(buffer), format, args);
+    drawText(dpi, x, y, paint, buffer);
+}
+
+static void
+drawStringCompat(rct_drawpixelinfo *dpi, sint32 x, sint32 y, rct_string_id format, void *args, uint8 colour,
```

Comment:
Not sure why these methods are called `drawString...`, and the other is `drawText()`. We should use either 'string' or 'text' for consistency.